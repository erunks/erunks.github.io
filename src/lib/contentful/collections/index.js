export { default as BUSINESS_CARD_FIELDS } from './business_card';
export { default as SOCIAL_FIELDS } from './social';
export { default as WORK_EXPERIENCE_FIELDS } from './work_experience';
