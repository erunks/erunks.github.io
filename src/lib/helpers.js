import assignIn from 'lodash/assignIn';
import get from 'lodash/get';
import join from 'lodash/join';
import split from 'lodash/split';

export const createVcfFromBusinessCard = (vCard, businessCardInfo) => {
  const keyMappings = {
    uid: get(businessCardInfo, 'uuid', ''),
    namePrefix: get(businessCardInfo, 'prefix', ''),
    firstName: get(businessCardInfo, 'firstname', ''),
    middleName: join(get(businessCardInfo, 'middlenames', []), ' '),
    lastName: get(businessCardInfo, 'lastname', ''),
    nameSuffix: get(businessCardInfo, 'suffix', ''),
    birthday: businessCardInfo?.birthday && new Date(businessCardInfo.birthday),
    organization: get(businessCardInfo, 'organization', ''),
    title: get(businessCardInfo, 'jobTitle', ''),
  };

  const logoUrl = get(businessCardInfo, 'logo.url', '');
  if (logoUrl) {
    vCard.logo.attachFromUrl(logoUrl, split(logoUrl, '.').pop());
  }

  return assignIn(
    vCard,
    keyMappings,
    get(businessCardInfo, 'emails', {}),
    get(businessCardInfo, 'faxNumbers', {}),
    get(businessCardInfo, 'phoneNumbers', {})
  );
};

export const downloadFile = (filename, contentType, data) => {
  const link = document.createElement('a');
  link.href = `data:${contentType};charset=utf-8,${data}`;
  link.download = filename;
  link.click();
};

export const handleKeyUp = (e, callback) => {
  // if keycode is enter or space
  if (e.keyCode === 13 || e.keyCode === 32) {
    callback();
  }
};

export default {
  createVcfFromBusinessCard,
  downloadFile,
  handleKeyUp,
};
