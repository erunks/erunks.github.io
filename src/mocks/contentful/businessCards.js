export const mockBusinessCards = [
  {
    uuid: '1',
    prefix: null,
    firstname: 'John',
    middlenames: ['A.'],
    lastname: 'Smith',
    birthday: '1990-01-01T00:00:00.000-04:00',
    addresses: null,
    phoneNumbers: {
      cellPhone: '555-123-4567',
    },
    faxNumbers: null,
    emails: {
      workEmail: 'john.smith@email.com',
    },
    logo: {
      url: 'https://example.com/icon.png',
    },
    organization: 'Example Company',
    jobTitle: 'A Position',
    socialUrls: {
      linkedIn: 'https://www.linkedin.com/in/johnasmith/',
    },
  },
];

export default mockBusinessCards;
