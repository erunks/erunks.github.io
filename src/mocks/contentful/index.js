export { default as mockBusinessCards } from "./businessCards";
export { default as mockSocialLinks } from "./socialLinks";
export { default as mockWorkExperiences } from "./workExperiences";
