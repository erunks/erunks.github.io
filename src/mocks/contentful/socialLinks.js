export const mockSocialLinks = [
  {
    icon: {
      url: 'https://www.placecage.com/32/32',
    },
    name: 'placecage',
    url: 'https://www.placecage.com/',
  },
];

export default mockSocialLinks;
