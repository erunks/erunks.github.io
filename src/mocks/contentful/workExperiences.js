export const mockWorkExperiences = [
  {
    altLogo: {
      url: 'https://example.com/icon.png',
    },
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    endDate: null,
    location: 'A Location',
    logo: {
      url: 'https://example.com/icon.png',
    },
    name: 'Example Company',
    position: 'A Position',
    startDate: '2018-01-01T00:00:00.000-05:00',
  },
]

export default mockWorkExperiences;
