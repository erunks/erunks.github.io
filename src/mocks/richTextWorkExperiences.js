export const richTextWorkExperiences = [
  {
    altLogo: { url: 'https://www.placecage.com/600/125' },
    description: {
      content: [
        {
          content: [
            {
              data: {},
              marks: [],
              nodeType: 'text',
              value:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
            },
          ],
          data: {},
          nodeType: 'paragraph',
        },
      ],
      data: {},
      nodeType: 'document',
    },
    endDate: null,
    locale: 'en',
    location: 'Geo, Location',
    logo: { url: 'https://www.placecage.com/600/125' },
    name: 'Company',
    position: 'Position',
    startDate: '2021-06-01T00:00:00.000-05:00',
  },
];

export default richTextWorkExperiences;
