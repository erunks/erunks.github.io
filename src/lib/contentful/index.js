import axios from 'axios';
import {
  CONTENTFUL_ACCESS_TOKEN,
  CONTENTFUL_PREVIEW_ACCESS_TOKEN,
  CONTENTFUL_SPACE_ID,
  IS_PRODUCTION,
} from 'lib/constants';
import get from 'lodash/get';
import join from 'lodash/join';
import map from 'lodash/map';
import { SOCIAL_FIELDS } from './collections/social';
import { WORK_EXPERIENCE_FIELDS } from './collections/work_experience';

export const stringifyQuery = (query) => {
  const replacer = (_key, value) => {
    if (typeof value === 'string') {
      return value.replace(/\s?\n\s*/g, '\n ');
    }
    return value;
  };

  const stringifiedQuery = JSON.stringify({ query }, replacer);

  return stringifiedQuery;
};

const postGraphQL = async (query, preview = IS_PRODUCTION) => {
  try {
    const url = `https://graphql.contentful.com/content/v1/spaces/${CONTENTFUL_SPACE_ID}`;
    const headers = {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${
          preview ? CONTENTFUL_PREVIEW_ACCESS_TOKEN : CONTENTFUL_ACCESS_TOKEN
        }`,
      },
    };

    const { data } = await axios.post(url, stringifyQuery(query), headers);

    return data;
  } catch (err) {
    console.error(err);
  }
};

export const getAllFromCollection = async (
  collectionName,
  fields = '',
  options = {},
  preview = IS_PRODUCTION
) => {
  const queryOptions = join(
    map(options, (value, key) => `${key}:${value}`),
    ', '
  );
  const queryOptionsString = queryOptions ? `(${queryOptions})` : '';

  const items = await postGraphQL(
    `query {
      ${collectionName}${queryOptionsString} {
        items {
          ${fields}
        }
      }
    }`,
    preview
  );

  return get(items, `data.${collectionName}.items`, []);
};

export const getAllSocialLinks = async (
  options = {},
  preview = IS_PRODUCTION
) => {
  const socials = await getAllFromCollection(
    'socialCollection',
    SOCIAL_FIELDS,
    options,
    preview
  );
  return socials;
};

export const getAllWorkExperiences = async (options = {}, preview = false) => {
  const experiences = await getAllFromCollection(
    'workExperienceCollection',
    WORK_EXPERIENCE_FIELDS,
    options,
    preview
  );
  return experiences;
};

export default {
  getAllSocialLinks,
  getAllWorkExperiences,
};
