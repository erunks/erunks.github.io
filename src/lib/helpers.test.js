import { mockBusinessCards } from 'mocks/contentful';
import * as helpers from './helpers';

describe('helpers', () => {
  describe('#createVcfFromBusinessCard', () => {
    const businessCard = mockBusinessCards[0];

    it('should create a vcf from a business card', () => {
      const vcf = helpers.createVcfFromBusinessCard(businessCard);
      expect(vcf).toMatchSnapshot();
    });

    describe('when there is a logo url present in the card info', () => {
      it('should return a vcf with a logo and type', () => {
        const vcf = helpers.createVcfFromBusinessCard(businessCard);
        expect(vcf.logo).toEqual(businessCard.logo);
      });
    });

    describe('when there is no logo url present in the card info', () => {
      it('should return a vcf with no logo', () => {
        const businessCardWithoutLogo = {
          ...mockBusinessCards[0],
          logo: {},
        };
        const vcf = helpers.createVcfFromBusinessCard(businessCardWithoutLogo);
        expect(vcf.logo).toEqual({});
      });
    });
  });

  describe('#downloadFile', () => {
    it('should download the file', () => {
      const link = {
        click: jest.fn(),
      };
      jest.spyOn(document, 'createElement').mockImplementation(() => link);

      const contentType = 'text/plain';
      const data = 'Plain text data';
      const filename = 'file.txt';

      helpers.downloadFile(filename, contentType, data);

      expect(link.href).toEqual(`data:${contentType};charset=utf-8,${data}`);
      expect(link.download).toEqual(filename);
      expect(link.click).toHaveBeenCalledTimes(1);
    });
  });

  describe('#handleKeyUp', () => {
    it('should call the callback when the enter key is released', () => {
      const callback = jest.fn();
      const event = {
        keyCode: 13,
      };
      helpers.handleKeyUp(event, callback);
      expect(callback).toHaveBeenCalledTimes(1);
    });

    it('should call the callback when the space key is released', () => {
      const callback = jest.fn();
      const event = {
        keyCode: 32,
      };
      helpers.handleKeyUp(event, callback);
      expect(callback).toHaveBeenCalledTimes(1);
    });

    it('should not call the callback when the key released is not enter or space', () => {
      const callback = jest.fn();
      const event = {
        keyCode: 0,
      };
      helpers.handleKeyUp(event, callback);
      expect(callback).toHaveBeenCalledTimes(0);
    });
  });
});
