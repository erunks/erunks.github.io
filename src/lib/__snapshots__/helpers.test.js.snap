// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers #createVcfFromBusinessCard should create a vcf from a business card 1`] = `
Object {
  "addresses": null,
  "birthday": undefined,
  "emails": Object {
    "workEmail": "john.smith@email.com",
  },
  "faxNumbers": null,
  "firstName": "",
  "firstname": "John",
  "jobTitle": "A Position",
  "lastName": "",
  "lastname": "Smith",
  "logo": Object {
    "url": "https://example.com/icon.png",
  },
  "middleName": "",
  "middlenames": Array [
    "A.",
  ],
  "namePrefix": "",
  "nameSuffix": "",
  "organization": "",
  "phoneNumbers": Object {
    "cellPhone": "555-123-4567",
  },
  "prefix": null,
  "socialUrls": Object {
    "linkedIn": "https://www.linkedin.com/in/johnasmith/",
  },
  "title": "",
  "uid": "",
  "uuid": "1",
}
`;
